rclc/src/rclc/executor.c

```diff
@@ -1433,6 +1433,10 @@ _rclc_execute(rclc_executor_handle_t * handle)
       case TIMER:
         // case TIMER_WITH_CONTEXT:
         rc = rcl_timer_call(handle->timer);
+        if (rc == RCL_RET_TIMER_CANCELED) {
+          break;
```

Comment:
Either we modify the return value to `RCL_RET_OK` , if the timer was cancelled - so it is no longer considered as an error: 
```
          rc = RCL_RET_OK;
          break;
```

or we update the check where `_rclc_execute`is called

here: executor.c:1700
https://github.com/ros2/rclc/blob/f8de341e2f9926d91af765d4587d4ca04954666a/rclc/src/rclc/executor.c#L1701

and here executor.c:1048
https://github.com/ros2/rclc/blob/f8de341e2f9926d91af765d4587d4ca04954666a/rclc/src/rclc/executor.c#L1749
to
```
 if ((rc != RCL_RET_OK) && (rc != RCL_RET_TIMER_CANCELLED)) {
```

Reply:
I see what you mean.
I missed the fact that line `rc = rcl_timer_call(handle->timer);` updates the `rc` variable and calling `break;` will simply return the same error.

I am wondering now why it fixed the issue i was having.

Reply:
I wondered too :)
But now the return value is `RCL_RET_OK` for this case and your use-case should work.